lib/lz4.c

```diff
@@ -602,18 +646,29 @@ LZ4_FORCE_INLINE int LZ4_compress_generic(
                 if (unlikely(forwardIp > mflimit)) goto _last_literals;
 
                 match = LZ4_getPositionOnHash(h, cctx->hashTable, tableType, base);
-                if (dict==usingExtDict) {
+                if (dictDirective == usingExtDictCtx) {
+                    if (match < (const BYTE*)source) {
```

Comment:
Can we invert these conditions to avoid repetition in the inner `else`?

Reply:
That means performing the `match > source` branch even when unnecessary, though.